### DISCUSSION TITLE
What's the best practices of accessing ceph over flaky network connection?

### DISCUSSION THREAD

**Discussion by enihcam** [ceph/ceph-csi#4181] [Q&A]
What's the best practices of accessing ceph over flaky network connection? For example, can I setup a local dm-cache binding ceph with a local SSD to buffer the I/O? Thanks.

**Comment by nixpanic** [comment]
A flaky network will usually be quite problematic. There is no guarantee that data has not been modified on an other system once the network comes back after a temporary interruption. In case the data was changed remotely, and something wrote to the local dm-cache device, a split-brain can happen. Someone needs to decide which side of the data is the right one.

Maybe it is possible to tune RBD for working with dm-cache and network interruptions. Or, for CephFS, you may want to look into FS-Cache (I don't know if CephFS supports that though). The best venue to ask about guidance and experience from others is at https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/ . The discussions in this GitHub project are for Ceph-CSI, which is _only_ a driver to provision/mount Ceph based storage.
